theseus/embodied/robot/link.py

```diff
@@ -60,6 +65,9 @@ def set_children(self, children: List[Any]):
     def set_ancestors(self, ancesotrs: List["Link"]):
         self._ancestors = ancesotrs
 
+    def set_angle_ids(self, angle_ids: List[int]):
```

Comment:
what is this supposed to do?

Reply:
This is the ids of non-fixed joints that a link depends on. For example, a 7-link robot arm, the second link only depends on joint 1 and joint2. This is used to improve the computational efficiency of jacobians.

Reply:
Compared to tracing manually back to the root, the computational complexity isn't really affected by caching the dependent joints, since you still need linear time to go through them.
Also, performance shouldn't be an issue since you have a factory pattern to generate an object that has all of the computational paths traced.

Reply:
> We should have a more explicit/descriptive name for this than `angle_ids`.

Now `angle_ids -> ancestor_active_joint_ids`.